### PR TITLE
added option to use gridded 2D topo data for geodetic_mb_calibration

### PR DESCRIPTION
**Pull request by afisc**
Until now the function `massbalance.mb_calibration_from_geodetic_mb()` did use the height/elevation data of a glaciers flowline(s) to calibrate the massbalance parameters. This is now also possible by using gridded/distributed data via the additional boolean argument `mb_2d`, which is useful for the usage of distributed models like IGM. 

- [x] Tests added/passed
- [x] Fully documented
- [x] Entry in `whats-new.rst`

### CODE REVIEW

**Comment by afisc** [comment]
added a test case which compares the calibrated 'melt_f' (Melt Factor) for the 1D(flowline) and 2D(gridded_data) calibration method

File: oggm/core/massbalance.py [hunk 1]
```diff
@@ -1433,6 +1433,7 @@ def mb_calibration_from_geodetic_mb(gdir, *,
                                     overwrite_gdir=False,
                                     use_regional_avg=False,
                                     override_missing=None,
+                                    mb_2d=False,
```
Comment (fmaussion) [inline oggm/core/massbalance.py:1436]: replace `mb_2d` with `use_2d_mb` (to clarify that this is a yes/no kwarg)

File: oggm/core/massbalance.py [hunk 2]
```diff
@@ -1479,6 +1480,8 @@ def mb_calibration_from_geodetic_mb(gdir, *,
         if the reference geodetic data is not available, use this value instead
         (mostly for testing with exotic datasets, but could be used to open
         the door to using other datasets).
+    mb_2d : bool
+        2D data is used for the mass balance calibration
```
Comment (fmaussion) [inline oggm/core/massbalance.py:1484]:
```suggestion
        Set to True if the mass balance calibration has to be done of the 2D mask 
        of the glacier (for fully distributed runs only).
```

File: oggm/core/massbalance.py [hunk 3]
```diff
@@ -1673,6 +1679,8 @@ def mb_calibration_from_scalar_mb(gdir, *,
         if a `mb_calib.json` exists, this task won't overwrite it per default.
         Set this to True to enforce overwriting (i.e. with consequences for the
         future workflow).
+    mb_2d : bool
+        2D data is used for the mass balance calibration
```
Comment (fmaussion) [inline oggm/core/massbalance.py:1683]: same as above

File: oggm/core/massbalance.py [hunk 4]
```diff
@@ -1741,7 +1749,18 @@ def mb_calibration_from_scalar_mb(gdir, *,
...
+        fls = gdir.read_pickle('inversion_flowlines')
+    else:
+        # if the 2D data is used, the flowline is not needed.
+        fls = None
+        # get the 2D data
+        fp = gdir.get_filepath('gridded_data')
+        with xr.open_dataset(fp) as ds:
+            # 'topo' instead of 'topo_smoothed'?
+            _heights = np.where(ds.glacier_mask, ds.topo_smoothed, np.nan).flatten()
+            heights = _heights[~np.isnan(_heights)]
```
Comment (fmaussion) [inline oggm/core/massbalance.py:1762]:
```      
_heights = np.where(ds.glacier_mask, ds.topo_smoothed, np.nan).flatten()
heights = _heights[~np.isnan(_heights)]
```

Can be replaced with:

```
heights = ds.topo_smoothed.data[ds.glacier_mask.data == 1]
```

**Review by fmaussion — COMMENTED** [review]
Thanks @afisc ! Great work.

A few comments attached.

Agreed with the problem "smoothed" vs not smoothed. Keep it as is for now (including the inline comment)

**Pull request merged**